### ISSUE TITLE
tokio-runtime-worker crash on raspberry pi 3 b+ raspberry pi os when trying to view site

### ISSUE THREAD

**Issue by SuddenPineapple** [mbrubeck/agate#33]
Everything works fine until I try to connect to the site when I get the error:
```
thread 'tokio-runtime-worker' panicked at 'removal index (is 0) should be < len (is 0)', library/alloc/src/vec.rs:1066:13
```

Also I am not sure whether or not this error is caused by agate but I will put it here. Occurred when installing with cargo on the same system:
```
error: unexpected `self` parameter in function
   --> /home/riley/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.2.0/src/net/tcp/stream.rs:627:40
    |
627 |         pub fn try_read_buf<B: BufMut>(&self, buf: &mut B) -> io::Result<usize> {
    |                                        ^^^^^ not valid as function parameter
    |
    = note: `self` is only valid as the first parameter of an associated function

   --> /home/riley/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.2.0/src/macros/cfg.rs:132:13
    |
131 |               #[cfg_attr(docsrs, doc(cfg(feature = "io-util")))]
    |                                                                 - expected one of 9 possible tokens
132 |               $item
    |               ^^^^^ unexpected token
    | 
   ::: /home/riley/.cargo/registry/src/github.com-1ecc6299db9ec823/tokio-1.2.0/src/net/tcp/stream.rs:569:5
    |
569 | /     cfg_io_util! {
570 | |         /// Try to read data from the stream into the provided buffer, advancing the
571 | |         /// buffer's internal cursor, returning how many bytes were read.
572 | |         ///
...   |
645 | |         }
646 | |     }
    | |_____- in this macro invocation

error: aborting due to 2 previous errors

error: could not compile `tokio`.
```

**Comment by Johann150** [comment]
Hi there, I have one more question: Did you install it with `cargo install agate` or did you clone the repository and run `cargo run`?

I suspect that this might be a problem with the new Tokio release 1.2.0, but not really sure why there should be a problem. I hope it is fairly unlikely that Agate itself causes a problem like this within Tokio.

**Comment by SuddenPineapple** [comment]
I used `cargo install agate`

**Comment by Johann150** [comment]
If along with the error message you got a hint like this: `note: Run with RUST_BACKTRACE=1 for a backtrace.` it would be helpful if you could run the same command, but prefixed with what the note says, i.e. like this `RUST_BACKTRACE=1 agate ...` and then the configuration options you used.
This would give some more information about where exactly the crash happened, and what led to it.

If you did not get such a message it would be good if you would instead clone the repository and try compiling from source, like this:
```bash
git clone https://github.com/mbrubeck/agate.git --depth=1 # depth limit speeds up cloning
cd agate # enter the directory git cloned to
RUST_BACKTRACE=1 cargo run -- <your options here> # compile and run agate
```
Of course replace `<your options here>` with the options you used on agate if you do this.

**Comment by mbrubeck** [comment]
I believe this is caused by this line when reading the key file (`key.rsa`):

https://github.com/mbrubeck/agate/blob/544f577b59c1a97336d110c87779dda3238b576e/src/main.rs#L189

We should check if `keys.is_empty()` before attempting this, so we can print a better error message.

The error might mean that your key file is not in the expected format.  Double-check that you are using the correct file, or try regenerating it with the command shown in the documentation.

> Also I am not sure whether or not this error is caused by agate but I will put it here.

What version of Rust are you using?  You can check with this command: `cargo --version`

**Comment by Johann150** [comment]
@mbrubeck indeed I get the same error when running Agate with an empty `key.rsa` file. Added an assertion before the line in question for now.

**Comment by Johann150** [comment]
I assume this has been resolved by 8fd9ca15c3df8757b0268f9be49a952142fb333f, and the problem was that your `key.rsa` file did not contain a valid key.

@SuddenPineapple to fix this with the latest published release, you should try regenerating your key and certificate file with the instructions from the README:
https://github.com/mbrubeck/agate/blob/49813d0c68137b24b70380068cc826742ef3b920/README.md#L30-L35
Maybe you could give feedback if that was indeed the issue here.
